extensions/amp-analytics/0.1/test/test-requests.js

```diff
@@ -168,6 +168,85 @@ describes.realWin('Requests', {amp: 1}, env => {
         expect(spy.args[0][0]).to.equal('r1&e1=e1&e2=e2&r2');
       });
     });
+
+    describe('batch plugin', () => {
+      it('should throw error when defined on non batched request', () => {
+        const spy = sandbox.spy();
+        const r = {'baseUrl': 'r', 'maxDelay': 0, 'batchPlugin': '_ping_'};
+        try {
+          new RequestHandler(ampdoc, r, preconnect, spy, false);
+        } catch (e) {
+          expect(e).to.match(
+              /batchPlugin cannot be set on non-batched request/);
```

Comment:
batchPlugin doesn't have an obvious relationship with maxDelay without having already read the docs for a user. Maybe change the error message to something like "Cannot use batchPlugin with a maxDelay of zero?"

Reply:
That is clear right now. But we are going to support batching interval later on. In that case user can still use batchPlugin with `maxDelay = 0` `batchInterval=...`. That's also why I am introducing `RequestHandler.isBatched_` in this PR.

Reply:
SGTM